src/blackdoor/cqbe/rpc/Rpc.java

```diff
@@ -39,6 +39,14 @@ public static Rpc fromJson(JSONObject rpcJson) throws RPCException {
 				return rpcObject;
 			case "GET":
 				rpcObject = new GetRpc();
+				GetRpc rpcGetObject = (GetRpc) rpcObject;
+				try {
+					rpcGetObject.index = rpcJson.getJSONObject("params")
+							.getInt("index");
+				}
+				catch (JSONException e){
+					throw new RPCException(JSONRPCError.INVALID_PARAMS);
+				}
```

Comment:
I think we're gonna want to do this kind of checking back in the Rpc class itself, in a constructor or something? @kag0 what say you?